CMakeLists.txt

```diff
@@ -0,0 +1,150 @@
+##################################################################
+# CMake file for Modelica Buildings Library
+# To build all targets, run
+# rm -rf build && mkdir build && cd build && cmake ../ && cmake --build . --target install && cd ..
+#
+##################################################################
+cmake_policy(SET CMP0048 NEW)
+
+project( Modelica-Buildings VERSION 5.1.0 )
+cmake_minimum_required(VERSION 3.8)
+
+# find_package(Java REQUIRED)
+# include(UseJava)
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g -ldl")
```

Comment:
it is unconventional to hard code `-g`. This should be specified as a config option, `CMAKE_BUILD_TYPE`, at configure time for make based generators, or as a build option at build time for multi config generators such as visual studio.

Linking to dl would be handled in a more cross platform way by linking to ${CMAKE_DL_LIBS} like this https://stackoverflow.com/questions/33678965/need-to-link-cmake-project-to-dl-library

Reply:
Thanks for the feedback, addressed through 40a3f55